it/test/uk/gov/hmrc/pillar2submissionapi/UktrSubmissionISpec.scala

```diff
@@ -0,0 +1,138 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package test.uk.gov.hmrc.pillar2submissionapi
+
+import org.scalatest.OptionValues.convertOptionToValuable
+import play.api.libs.json.Json
+import play.api.mvc.AnyContentAsJson
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import test.uk.gov.hmrc.pillar2submissionapi.UktrSubmissionISpec._
+import uk.gov.hmrc.http.HttpVerbs.POST
+import uk.gov.hmrc.pillar2submissionapi.base.IntegrationSpecBase
+import uk.gov.hmrc.pillar2submissionapi.controllers.routes
+
+class UktrSubmissionISpec extends IntegrationSpecBase {
+
+  "UKTR Submission" when {
+    "Creating a new UKTR submission (POST)" that {
+      "has valid submission data" should {
+        val request = FakeRequest(POST, routes.UktrSubmissionController.submitUktr.url)
+          .withBody[AnyContentAsJson](validRequestJson)
+
+        "return a 201 CREATED response" in {
+          val application = applicationBuilder().build()
+          running(application) {
+            val result = route(application, request).value
+            status(result) mustEqual CREATED
+          }
+        }
+      }
+      "has valid nil-return submission data" should {
+        val request = FakeRequest(POST, routes.UktrSubmissionController.submitUktr.url)
+          .withBody[AnyContentAsJson](validRequestNilReturnJson)
+
+        "return a 201 CREATED response" in {
+          val application = applicationBuilder().build()
+          running(application) {
+            val result = route(application, request).value
+            status(result) mustEqual CREATED
+          }
+        }
+      }
+      "has an invalid request body" should {
+        val request = FakeRequest(POST, routes.UktrSubmissionController.submitUktr.url)
+          .withBody[AnyContentAsJson](invalidRequestJson)
+
+        "return a 400 BAD_REQUEST response" in {
+          val application = applicationBuilder().build()
+          running(application) {
+            val result = route(application, request).value
+            status(result) mustEqual BAD_REQUEST
+          }
+        }
+      }
+      "has no request body" should {
+        val request = FakeRequest(POST, routes.UktrSubmissionController.submitUktr.url)
+
+        "return a 400 BAD_REQUEST response" in {
+          val application = applicationBuilder().build()
+          running(application) {
+            val result = route(application, request).value
+            status(result) mustEqual BAD_REQUEST
+          }
+        }
+      }
+    }
+
+    "Amend a UKTR submission (PUT)" that {
+      "has a valid request body" should {
+        "return a 200" in {
+          // Not yet implemented
+          true
+        }
+      }
+    }
+  }
+}
+
+object UktrSubmissionISpec {
+  val validRequestJson: AnyContentAsJson =
+    AnyContentAsJson(Json.parse("""{
```

Comment:
this `AnyContentAsJson` wrapping seems a little odd.

With some import wizardry, you can just use Json.parse... and inject a JsValue into a request.

Remove the explicit `Json` import and import `import play.api.libs.json._` instead and you can remove this wrapping

Reply:
A little unsure on this one... I've added the import you'd suggested, but even with that it then complains that I'm now returning the wrong type
<img width="732" alt="image" src="https://github.com/user-attachments/assets/6d77878f-c17f-4ae5-adc7-cbc8594158ff">
Surely adding this would mean I could just remove the Json. from the Json.parse(XXX) part?

Reply:
Probably also worth mentioning that I'd not seen another POST request in other repo's that we test with a full request body provided. The closest one looked like this:
![image](https://github.com/user-attachments/assets/dd5f7d94-ae96-4078-8341-a3b565229e5e)
so this was just by best attempt! :D